### ISSUE TITLE
Add/configure facets

### ISSUE THREAD

**Issue by bkiahstroud** [harvard-lts/CURIOSity#138]
## Summary

No facets are appearing on the left filter rail when conducting a search or in the Search > Facets tab in the admin dashboard

![Image](https://user-images.githubusercontent.com/32469930/184038658-ddfb5fe1-92c7-4418-9ece-3d3cee601112.png)

## Acceptance Criteria

- These fields show up as facets: 
  - [x] Creator / Contributor (We currently have this pluralized, but we'd like this to display as singular if possible)
  - [x] Place of Origin
  - [x] Publisher
  - [ ] Date (We currently have Creation Date, but we'd like this to display as Date if possible)
  - [x] Language
  - [x] Genre
  - [x] Culture
  - [x] Style / Period
  - [x] Material / Technique (We currently have this pluralized, but we'd like this to display as singular if possible)
  - [x] Subject (We currently have this pluralized, but we'd like this to display as singular if possible)
  - [x] Curator Tag (Rename the Exhibit Tags field to this)
  - [x] Place (We currently have this pluralized, but we'd like this to display as singular if possible)
  - [x] Series
  - [x] Collection
  - [x] Digital Format
  - [x] Repository
  - [x] Available To (This one is the only one that should be UNchecked for display, all others should be checked)
- [ ] When importing exhibits, ensure imported facets work 

## Testing Instructions

1. Login as an admin 
2. Create a new exhibit 
3. Navigate to Exhibit Dashboard > Items > Add Item  
4. Fill out the fields to harvest the `demo` set (use [this mapping file](https://github.com/harvard-lts/spotlight-oaipmh-resources/blob/job_entry/harvard_yaml_mapping_files/mods/curiosity.yml))  
5. Click Add MODS Items 
6. Click the magnifying lens next to the search bar 
7. Validate the facets 
8. Navigate to Exhibit Dashboard > Search > Facets 
9. Validate the facets 

## Notes

Facets need to be setup in the catalog controller

**Comment by bkiahstroud** [comment]
@sephirothkod how do we determine which fields should be faceted? Do we have that info?

**Comment by sephirothkod** [comment]
Their current implementation uses:

- Date
- Type
- Language
- Repository

I do not believe we have been given any info as to whether or not they want to change/add any facets though.

**Comment by sephirothkod** [comment]
@bkiahstroud I messed up. Here are the actual fields that they have in their current implementation:



![Image](https://user-images.githubusercontent.com/6759631/184070203-e34681e9-a0bb-46d0-a703-66032c46cc44.png)

**Comment by bkiahstroud** [comment]
No worries! I've updated the Acceptance Criteria

**Comment by VanessaVenti** [comment]
Hi @bkiahstroud - I don't think I ever gave you a proper list for what we'd like our facets to display as. The list you created in the acceptance criteria was close, but I just made some updates now which should be our official list. I did notice that what you had before didn't quite match what I was seeing in QA so it looks like it needs some updates anyway :).

**Comment by bkiahstroud** [comment]
@VanessaVenti thank you for updating the list! 

@summer-cook does this need to go back to In Development? Do those fields need to be explicitly added to the code?

**Comment by summer-cook** [comment]
@bkiahstroud  Yes- they do need to be explicitly added 

One thing I am noticing is this:
- [ ] Available To (This one is the only one that should be UNchecked for display, all others should be checked)

I'm not sure there is a way to make it default to being unchecked for display if we add it as a facet in the catalog controller. This might take some extra work

**Comment by bkiahstroud** [comment]
@VanessaVenti would you mind clarifying what you mean by "[Available to] is the only one that should be UNchecked for display"? What should that look like in the UI?

**Comment by VanessaVenti** [comment]
@summer-cook and @bkiahstroud - For unchecking Available To to display, if there is not an easy way to have a facet unchecked by default, that's ok, no need to spend extra time on it. That's a facet that we currently harvest the data for but don't display to end users, but might at some point which is why we pull it in. If the metadata fields for the different views work differently and we are able to set up which ones display/don't by default, those would be more important. Happy to discuss when we meet this afternoon! 

Also, can the order displayed in acceptance criteria be set as the default?

**Comment by VanessaVenti** [comment]
Here's a screenshot of what it would look like in the UI where under Search > Facets, all would be checked by default aside from Available To:

![Screen Shot 2022-08-23 at 1 17 06 PM](https://user-images.githubusercontent.com/60190174/186223329-7fd906e4-43ae-4fc6-8a76-af8c543b5841.png)

**Comment by bkiahstroud** [comment]
From Vanessa: 

If possible, facets should display as singular (e.g. "Subject"), and the same field will show up as plural elsewhere (i.e. item show page, e.g. "Subjects")

**Comment by bkiahstroud** [comment]
@VanessaVenti I see two date-related fields: `creation-date_ssim` and `date_ssim`. 

To clarify, you're saying we're currently faceting the former when we should be faceting the latter? Or just that the former's label needs to be changed?

**Comment by VanessaVenti** [comment]
@bkiahstroud For Available To, yes I just meant to have that check box unchecked by default and then when/if we decide to show it we can switch it to checked.

For creation-date_ssim vs date_sssim - we have both in our yaml because Date was a default (baked-in) field in Spotlight that we never figured out how to make it work so we made Creation Date and use that instead. We just need one ideally we would call it Date. This is one of those examples where the baked-in/default fields worked differently than the fields we created so we just made a new one instead of using the one that already existed. I made notes about all the odd ones like this in the [2.0 Metadata Template spreadsheet.](https://docs.google.com/spreadsheets/d/1ku4jp73zvvFLSJt0j80h53g3Z7SThn-3rcFOPLpesk8/edit#gid=0)

For Exhibit Tag - this is another Spotlight default field so we don't have it in our yaml. It looks like it's working fine, we just want to change the field name to Curator Tag. We don't use this field currently so you won't see it in any of our live exibits, but we plan to use it in this release. 

Do let me know if anything is still unclear, more than happy do do a quick zoom anytime if that is helpful!

**Comment by summer-cook** [comment]
### Attempted to QA

This is on a new exhibit that has the contagion set harvested
These are the only facets that are imported

![Image](https://user-images.githubusercontent.com/73361970/187492771-cf63bb7d-648f-43bd-b5c3-57f441feac18.png)

oddly when I search with FTS all the facets show up


![Image](https://user-images.githubusercontent.com/73361970/187492485-c49e4628-1036-4172-8f33-e6fd7b640c96.png)

**Comment by bkiahstroud** [comment]
@summer-cook The facets only appear if there are items that have data for them. I've added testing instructions to clarify

**Comment by summer-cook** [comment]
@bkiahstroud i believe they should still appear in the exhibit dashboard (first screenshot above) though, even if there are no items with that facet- I see other exhibits that have the facets listed in that part of the dashboard, and the facets say they have zero items.

Here's an example:
https://staging-harvard-curiosity.notch8.cloud/test/metadata_configuration/edit

So I think something else is going on where the facetable metadare not importing or being added correctly anymore

**Comment by summer-cook** [comment]
@bkiahstroud also, i just tried to follow the testing instuctions.

I created an item that had all the fields filled
(I noticed in the manual item creation form there are A TON of fields- way more than are on the list of fields.)
As soon as I hit add item, I tried to view all items and now the app throws an error: https://staging-harvard-curiosity.notch8.cloud/summer-contagion/catalog/admin

**Comment by bkiahstroud** [comment]
@summer-cook I set up an exhibit / [item](https://staging-harvard-curiosity.notch8.cloud/i138-facets/catalog/20-990006049270203941_RADSCHL:1267673) to test this

**Comment by bkiahstroud** [comment]
> @bkiahstroud i believe they should still appear in the exhibit dashboard (first screenshot above) though, even if there are no items with that facet- I see other exhibits that have the facets listed in that part of the dashboard, and the facets say they have zero items.
>
>Here's an example:
>https://staging-harvard-curiosity.notch8.cloud/test/metadata_configuration/edit
>
>So I think something else is going on where the facetable metadata are not importing or being added correctly anymore

@summer-cook that is because those were "custom" facets. They have since been replaced with built-in default facets that behave slightly differently (i.e. not showing up unless data exists in them)

**Comment by summer-cook** [comment]
@bkiahstroud 

Ok just retested this
**Two things I found:**

> Available To (This one is the only one that should be UNchecked for display, all others should be checked)
- The "Available To" facet does not show up at all in the exhibit dashboard. Is it supposed to show up, but as unchecked? Or is not showing up what is supposed to happen

- the facet "Collection" in the list above is labeled "In Collection" instead - not sure if this is intended. 


Other than these 2 questions, all other facets from the list above appear, so everything else passes QA.

**Comment by VanessaVenti** [comment]
@summer-cook I think I can help answer these:

- 'Available To' - ideally we'd like it to show up as a facet and but be unchecked by default (but with the option to check it should we decide to display it)
- 'Collection' should be labeled as such for faceting (ssim). There is a separate 'In Collection' free text (tesim) field.

**Comment by dl-maura** [comment]
@VanessaVenti this is going to QA today, it looks right to Phil and I but I'd like your sign off as you know the fields best

**Comment by VanessaVenti** [comment]
@bkiahstroud - I'm still reviewing this one, but so far I noticed that Subject is not pulling in the same metadata.

Here is what I see in 2.0 for the demo set:
![Image](https://user-images.githubusercontent.com/60190174/188968444-2c66cfc5-3679-439a-9504-0a6579fc95b0.png)

Versus 3.0:
![Image](https://user-images.githubusercontent.com/60190174/188968515-5433cf62-fdc1-47cc-9449-b0a2fe4f2adc.png)

**Comment by VanessaVenti** [comment]
@bkiahstroud I'm also seeing slightly different totals for Repository where I think 3.0 is only pulling in the first occurrence of a Repository in the facets for an item and 2.0 is pulling in all occurrences. 

2.0
![Image](https://user-images.githubusercontent.com/60190174/188970056-05360df2-acc0-449d-9d2f-3cf2ed5afbd9.png)

3.0 (see Houghton Library facet as an example)
![Image](https://user-images.githubusercontent.com/60190174/188970247-2c9aade0-024d-4fe5-8d20-56c94e014313.png)

**Comment by VanessaVenti** [comment]
@bkiahstroud regarding your notes above:

1. "Exhibit Tags" is a special field in Spotlight, so changing the label to say "Curator Tag" is a different and more involved process compared to the other labels. It's possible, but is probably big enough to be its own ticket in my estimation (The same goes for its order in the list)
- Let's break this one out into its own ticket then if you haven't already. @enriquediaz and @crisr15 Tagging is a new feature we are turning on in 3.0, but something BTA and other curators have wanted for a while so I'm not sure which team makes sense to tackle this one. Changing the label name is something I checked in with Amy D. about for accessibility as Exhibit Tags seemed like a confusing label, but I realize this is likely messing with Spotlight core so let me know if this is a bad idea or just not worth the effort.

2. Excluding Exhibit Tags, the default labels and order have been configured to match what's in the Acceptance Criteria. However, if an Exhibit is imported that had different labels and/or order for the facets, the imported values will override the defaults.
- Do you think everything would map properly if we went and changed all the labels in the 2.0 exhibits (without reharvesting) to what we want them to be for 3.0? I'm happy to test that and do that for all if you think that is the best option.

**Comment by bkiahstroud** [comment]
Here's the ticket to change the Exhibit Tags label: #188 

>Do you think everything would map properly if we went and changed all the labels in the 2.0 exhibits (without reharvesting) to what we want them to be for 3.0? I'm happy to test that and do that for all if you think that is the best option.

It's worth testing, but I think that would work and probably be the best option. It shouldn't require any reharvesting in the old app or new

**Comment by VanessaVenti** [comment]
@bkiahstroud  All looks like it is working for new exhibits, but we found for imported 2.0 exhibits Date appears twice, once with items, once without:
![Image](https://user-images.githubusercontent.com/60190174/189701274-de0dc1a1-935c-4e44-96d6-991758a0268d.png)

This is the json file I used for testing: https://drive.google.com/file/d/1GlgppD-KwvgJK20bZn6nQRk5V_7Nt9sf/view?usp=sharing

I updated https://drive.google.com/file/d/1dSeBza_TKPRZZigxloL7x8FiGD_Q2njx/view?usp=sharing with our 3.0 labels so I could test that one as well, but I'm unable to import that on into our QA at the moment. I'll keep you posted on that.

**Comment by bkiahstroud** [comment]
@VanessaVenti I believe this is because the custom date field is being carried over from the old exhibit in the JSON. Does one of the date fields show up under the "Exhibit-specific fields" section? If so, you should just be able to delete it there.

**Comment by bkiahstroud** [comment]
@VanessaVenti also, that reminds me, there are some changes that need to be made to the curiosity.yml mapping file. We were able to get the default fields to work after all, so these have all been removed: 

- `creation-date_tesim` (use `date_tesim` instead) 
- `spotlight_upload_date_tesim` (use `date_tesim` instead)
- `spotlight_upload_attribution_tesim` (use `attribution_tesim` instead)

**Comment by bkiahstroud** [comment]
There may be some more changes to make in the mapping files coming soon, but these are the ones so far.

**Comment by dl-maura** [comment]
Is this file in version control anywhere @VanessaVenti ?

**Comment by dl-maura** [comment]
if we're keeping it there @VanessaVenti then we'll want to commit those changes there as well

**Comment by dl-maura** [comment]
I dont see 
spotlight_upload_date_tesim (use date_tesim instead)
spotlight_upload_attribution_tesim (use attribution_tesim instead)

in the yml?

**Comment by bkiahstroud** [comment]
>I dont see
>spotlight_upload_date_tesim (use date_tesim instead)
>spotlight_upload_attribution_tesim (use attribution_tesim instead)
>
>in the yml?

That's not a problem, I didn't check all the yml files, just listed all the changes to the fields so far in case they were there

**Comment by VanessaVenti** [comment]
@bkiahstroud - For the duplicate Date facet issue - yes, I see I have Creation Date as an exhibit specific field and deleting that clears it in the Search > Facets list. Thanks! @dl-maura I added this as a step in https://github.com/harvard-lts/CURIOSity/issues/88 when we get there. 

I'm going to go ahead and close this one, @dl-maura I created [a new ticket to track curiosity.yml updates](https://github.com/harvard-lts/CURIOSity/issues/199) where I put the info in the comments above
